web/src/beta/lib/reearth-ui/components/Selector/index.tsx

```diff
@@ -0,0 +1,252 @@
+  disabled,
+  onChange,
+}) => {
+  const theme = useTheme();
+  const selectorRef = useRef<HTMLDivElement>(null);
+  const [selectedValue, setSelectedValue] = useState<string | string[] | undefined>(
+    value ?? (multiple ? [] : undefined),
+  );
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [selectorWidth, setSelectorWidth] = useState<number>();
+
+  const optionValues = useMemo(() => options, [options]);
+
+  useEffect(() => {
+    setSelectedValue(value ?? (multiple ? [] : undefined));
+  }, [value, multiple]);
+
+  const updateSelectorWidth = useCallback(() => {
+    if (selectorRef.current) {
+      const newWidth = selectorRef.current.clientWidth;
+      setSelectorWidth(newWidth);
+      console.log("Selector width updated:", newWidth);
+    }
+  }, []);
+
+  useEffect(() => {
+    updateSelectorWidth();
+    window.addEventListener("resize", updateSelectorWidth);
+    return () => {
+      window.removeEventListener("resize", updateSelectorWidth);
+    };
+  }, [updateSelectorWidth]);
+
+  useEffect(() => {
+    if (selectorRef.current) {
+      const observer = new ResizeObserver(updateSelectorWidth);
+      observer.observe(selectorRef.current);
+      return () => {
+        observer.disconnect();
+      };
+    }
+    return () => {};
+  }, [selectorRef, updateSelectorWidth]);
+
+  const isSelected = useCallback(
+    (value: string) => {
+      if (multiple) {
+        return Array.isArray(selectedValue) && selectedValue.includes(value);
+      }
+      return selectedValue === value;
+    },
+    [multiple, selectedValue],
+  );
+
+  const handleChange = (value: string) => {
+    if (multiple && Array.isArray(selectedValue)) {
+      if (selectedValue.includes(value)) {
+        const newSelectedArr = selectedValue.filter(val => val !== value);
+        setSelectedValue(newSelectedArr);
+        onChange?.(newSelectedArr);
+      } else {
+        const newSelectedArr = [...selectedValue, value];
+        setSelectedValue(newSelectedArr);
+        onChange?.(newSelectedArr);
+      }
+    } else {
+      if (value === selectedValue) setSelectedValue(undefined);
+      else setSelectedValue(value);
+      setIsOpen(!isOpen);
+      onChange?.(value);
+    }
+  };
+
+  const handleUnselect = useCallback(
+    (e: MouseEvent<HTMLElement>, value: string) => {
+      e.stopPropagation();
+      if (Array.isArray(selectedValue) && selectedValue.length) {
+        const newSelectedArr = selectedValue.filter(val => val !== value);
+        setSelectedValue(newSelectedArr);
+        onChange?.(newSelectedArr);
+      }
+    },
+    [selectedValue, onChange],
+  );
+
+  const renderTrigger = () => {
+    return (
+      <SelectInput isMultiple={multiple} isOpen={isOpen} disabled={disabled} width={selectorWidth}>
+        {!selectedValue?.length ? (
+          <Typography size="body" color={theme.content.weaker}>
+            {placeholder}
+          </Typography>
+        ) : multiple ? (
+          <SelectedItems>
```

Comment:
Please doube comfirm with Red, currently multiple selected item will make the Selector higher, which i think should not happen (when it's still single line).